### ISSUE TITLE
Adding "-v" flag to drive-examples command hangs

### ISSUE THREAD

**Issue by cyanglaz** [flutter/flutter#69797]
Steps to reproduce:
1. Apply this diff to https://github.com/flutter/plugin_tools/blob/master/lib/src/drive_examples_command.dart
```git
diff --git a/lib/src/drive_examples_command.dart b/lib/src/drive_examples_command.dart
index 59c6422..8b1fa36 100644
--- a/lib/src/drive_examples_command.dart
+++ b/lib/src/drive_examples_command.dart
@@ -120,7 +120,7 @@ Tried searching for the following:
             }
           }
 
-          final List<String> driveArgs = <String>['drive'];
+          final List<String> driveArgs = <String>['drive', '-v'];
 
           final String enableExperiment = argResults[kEnableExperiment];
           if (enableExperiment.isNotEmpty) {
```
2. Activate flutter_plugin_tools using local path:
`pub activate global -s path <path_to_plugin_tools_dir>`

3. In flutter/plugins directory, run `./script/incremental_build.sh drive-examples`, during running the drive test for the first plugin, the process hang at 
```
[ +100 ms] Running "flutter pub get" in example...
[   +4 ms] Using /Users/ychris/flutter/flutter/.pub-cache for the pub cache.
[   +1 ms] executing: [/Users/ychris/flutter/plugins/packages/android_alarm_manager/example/] /Users/ychris/flutter/flutter/bin/cache/dart-sdk/bin/pub --verbose get --no-precompile
[ +107 ms] MSG : Resolving dependencies...
```

After some investigation, it is caused by https://github.com/flutter/plugin_tools/blob/master/lib/src/drive_examples_command.dart#L150
If we replacing this method with https://github.com/flutter/plugin_tools/blob/master/lib/src/common.dart#L397, the hang is fixed.

**Comment by blasten** [comment]
@cyanglaz can you still repro the issue if you run `flutter drive -v` from the console, directly?

**Comment by cyanglaz** [comment]
> @cyanglaz can you still repro the issue if you run `flutter drive -v` from the console, directly?

no. It is the issue with the plugin_tools. `flutter drive -v` works.

**Comment by blasten** [comment]
that's strange. Have you been able to create a minimal repro with just `runAndStream`, and another Dart program that finishes after some wait time?

**Comment by cyanglaz** [comment]
@blasten I haven't looked into it too much yet, all I know right now is that it is failing for `flutter drive -v`.
